pkg/gateway/linkid.go

```diff
@@ -31,13 +31,20 @@ func (g *gateway) isAllowed(conn *proxyproto.Conn, host string) (string, bool, e
 		return "", false, err
 	}
 
+	return g.gatewayVerification(host, linkID)
```

Comment:
We can avoid spliting `isAllowed` (it is not too big) into smaller functions by changing this:

```go
linkID, err := linkID(conn)
```

to this:

```go
linkID, err := g.linkID(conn)
```

And defining `linkID` on `gateway` struct as follows:

```go
linkID func(conn *proxyproto.Conn) (string, error)
```

so instantiation will be like this:

```go
g := &gateway{
	// ...

	// note that current definiton of linkID function stays untouched.
	linkID: linkID,
}
```

This way we make it mockable from the test: we can replace it with an anonymous functions inside of the test which returns whatever we need.

And `linkID` can be tested separately.

Reply:
Just to clarify - not asking to change. This is just a tip. We do similar things when we want to mock something like `time.Now()`. [Here](https://github.com/Azure/ARO-RP/blob/4dbe573e026a736eaa39281f1a115f1d7a3b0a68/pkg/util/clientauthorizer/arm_test.go#L155) is an example of mocking `time.Now()` and [on the next line](https://github.com/Azure/ARO-RP/blob/4dbe573e026a736eaa39281f1a115f1d7a3b0a68/pkg/util/clientauthorizer/arm_test.go#L156-L168) we mock `http.Client`'s `Do` method. But it is not limited to standard library. I just can't remember of examples where we mock our own code this way.

I'll leave it up to you to decision how you want to approach it. IMO - splitting is fine in this case.